charts/external-es-proxy/templates/external-es-proxy-deployment.yaml

```diff
@@ -0,0 +1,109 @@
+#################################
+## Elasticsearch Proxy Deployment
+#################################
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "external-es-proxy.fullname" . }}
+  labels:
+    tier: external-logging
+    component: {{ template "external-es-proxy.name" . }}
+    chart: {{ template "external-es-proxy.chart" . }}
+    heritage: {{ .Release.Service }}
+    release: {{ .Release.Name }}
+spec:
+  {{- if not .Values.autoscaling.enabled }}
+  replicas: {{ .Values.replicaCount }}
+  {{- end }}
+  selector:
+    matchLabels:
+      tier: external-logging
+      component: {{ template "external-es-proxy.name" . }}
+      release: {{ .Release.Name }}
+  template:
+    metadata:
+      {{- with .Values.podAnnotations }}
+      annotations:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      labels:
+        tier: external-logging
+        component: {{ template "external-es-proxy.name" . }}
+        release: {{ .Release.Name }}
+      annotations:
+        checksum/nginx-config: {{ include (print $.Template.BasePath "/external-es-proxy-configmap.yaml") . | sha256sum }}
+        checksum/lua-config: {{ include (print $.Template.BasePath "/external-es-proxy-env-configmap.yaml") . | sha256sum }}
+    spec:
+      {{- with .Values.imagePullSecrets }}
+      imagePullSecrets:
+        {{- toYaml . | nindent 8 }}
+      {{- end }}
+      serviceAccountName: {{ include "external-es-proxy.serviceAccountName" . }}
+      securityContext:
+        {{- toYaml .Values.podSecurityContext | nindent 8 }}
+      containers:
+        - name: {{ .Chart.Name }}
+          securityContext:
+            {{- toYaml .Values.securityContext | nindent 12 }}
+          image: "{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}"
+          imagePullPolicy: {{ .Values.image.pullPolicy }}
+          env:
+{{- if .Values.global.custom_logging.secret }}
+          - name: ES_SECRET
+            value: {{.Values.global.custom_logging.secret }}
+{{- else if .Values.global.custom_logging.secretName }}
+          - name: ES_SECRET_NAME
+            valueFrom:
+              secretKeyRef:
+                name: {{.Values.global.custom_logging.secretName }}
+                key: elastic
```

Comment:
@danielhoherd - we need proper naming for this

Reply:
Maybe we use `envFrom` to populate `ES_SECRET`, then we only have a single env var to consider? Might also remove the need for the lua?

Reply:
we added lua because - nginx doesn't support direct env read in nginx config , and the value we pass inside the auth is base64, to handle these too we added lua.